iroha/src/sumeragi.rs

```diff
@@ -932,6 +923,25 @@ pub mod message {
                 signature: Signature::new(*public_key, &transaction_hash, private_key)?,
             })
         }
+
+        /// Checks that this `TransactionReceipt` is valid.
+        pub fn is_valid(&self, network_topology: &InitializedNetworkTopology) -> bool {
+            network_topology
+                .verify_signature_with_role(
+                    self.signature.clone(),
+                    Role::Leader,
+                    &self.transaction_hash,
+                )
+                .is_ok()
+        }
+
+        /// Checks if the block should have been already created by the `Leader`.
+        pub fn block_should_be_created(&self, block_time: Duration) -> bool {
```

Comment:
Let's rename this function - idk what option is good, something like `is_already_created` or another "question" form name.